### PR TITLE
bugfix: memory leak in ImageToBlob

### LINKED ISSUE #9: Memory Leak
I'm processing about 2000 images per minute across 8 nodes in production and noticed that system memory is going up.

:erlang.memory() is stable, but system memory goes up by about ~50mb per hour per node until the VM just crashes.

It seems that the NIF bindings from exmagick might be the issue here where memory is not being freed up somewhere.

The commands I'm using are init! image_load!, image_dump!, and size!

0.0.5

### PR DESCRIPTION
**Pull request by dgvncsz0f**
fixes #9 

image_dump!/1 is leaking memory because we are not deallocating memory
after `ImageToBlob`. the following script should exhaust the machine's
memory after a short while:

    ```
    defmodule Loop do
      def loop() do
        ExMagick.init!()
        |> ExMagick.image_load!("/tmp/foobar.tiff") # 16MB file
        |> ExMagick.image_dump!()
        loop()
      end
    end
    Loop.loop()
    ```

the solution is to copy the blob into a new new managed binary. this
might impact performance a bit depending on the size of the image but as far
as we know this can't be helped. it seems there is no way to attach a
destructor to a nif term other than `enif_open_resource_type` which
can't occur outside of `load` or `upgrade`.

### CODE REVIEW

**Comment by soteras** [comment]
Please if possible add some unit tests.

**Comment by dgvncsz0f** [comment]
@soteras i don't know how to add tests checking for possible memory leaks. do you have anything in mind?

File: mix.exs [hunk 1]
```diff
@@ -38,7 +29,7 @@ defmodule ExMagick.Mixfile do
     [
       app: :exmagick,
       version: "1.0.0",
-      elixir: "~> 1.2",
+      elixir: "~> 1.3",
```
Comment (soteras) [inline mix.exs]: What do you thing about update elixir to last version?

https://elixir-lang.org/blog/2018/01/17/elixir-v1-6-0-released/

**Comment by soteras** [comment]
@dgvncsz0f I don't know. One possibility is load some images and check the memory. is there a way to check memory of the server and others data?

File: mix.exs (hunk 1 above)
Comment (mememori) [reply mix.exs]: This is the minimum requirement.

There's no reason for us to require the most recent elixir version.

We can, however, do as we did here to require a version with OTP/20 to use the dirty scheduler

**Comment by dgvncsz0f** [comment]
you could use valgrind. but i think that will be quite hard to get right, as we would need to run it through elixir which forks/exec a bunch of executables. 

i would need to spend some time studying how to this reliably, but initially i don't know a good solution to include these kind of tests.

**Pull request merged**